app/engine/RowingStatistics.js

```diff
@@ -65,8 +65,9 @@ function createRowingStatistics (config, session) {
     // This is the core of the finite state machine that defines all state transitions
     switch (true) {
       case (sessionStatus === 'Paused' && rower.strokeState() === 'Drive'):
-        sessionStatus = 'Rowing'
+        // the order these are done in matters
         resumeTraining()
+        sessionStatus = 'Rowing'
```

Comment:
This feels a bit odd, as we are setting and resetting the same variable here. I think we have take a much closer look at the statemachine here, and see how it behaves in a paused scenario.

I see that we call resumeTraining() twice, once from the state machine, once as a response to an external event. I think we should add the allowResumeTraining() function, with the same content as the current resumeTraining(). Then we remove the line sessionStatus = 'WaitingForDrive' from the resumeTraining().

Reply:
I changed your code accordingly, could you test if that works well?